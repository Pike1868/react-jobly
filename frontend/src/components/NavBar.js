import React from "react";
import { Link as RouterLink } from "react-router-dom";
import { AppBar, Toolbar, Typography, Button, Link } from "@mui/material";
import { useNavigate } from "react-router-dom";
import { useUserContext } from "../context/UserContext";

const NavBar = () => {
  const { user, setUser } = useUserContext();
  const navigate = useNavigate();
  const logout = () => {
    localStorage.removeItem("joblyToken");
    setUser(null);
    navigate("/");
  };
  return (
    <AppBar
      position="static"
      sx={{
        display: "flex",
        flexDirection: "row",
        justifyContent: "space-between",
        backgroundColor: "#253031",
        maxWidth: "100%",
      }}
    >
      <Link component={RouterLink} to="/" style={{ textDecoration: "none" }}>
        <Typography
          variant="h4"
          component="div"
          sx={{ margin: "1rem", fontWeight: "700", color: "#a2bd9c" }}
          href="/"
        >
          JOBLY
        </Typography>
      </Link>
      <Toolbar
        sx={{
          display: "flex",
          justifyContent: "space-around",
        }}
      >
        <Link
          component={RouterLink}
          to="/companies"
          color="inherit"
          style={{ margin: "0 10px" }}
        >
          <Button color="inherit" style={{ fontWeight: "600" }}>
            Companies
          </Button>
        </Link>
        <Link
          component={RouterLink}
          to="/jobs"
          color="inherit"
          style={{ margin: "0 10px" }}
        >
          <Button color="inherit" style={{ fontWeight: "600" }}>
            Jobs
          </Button>
        </Link>
        <Link
          component={RouterLink}
          to="/profile"
          color="inherit"
          style={{ margin: "0 10px" }}
        >
          <Button color="inherit" style={{ fontWeight: "600" }}>
            Profile
          </Button>
        </Link>
        {user && (
          <Button
            color="inherit"
            style={{ fontWeight: "600", margin: "0 10px" }}
            onClick={logout}
          >
            Log Out
          </Button>
        )}
      </Toolbar>
    </AppBar>
  );
};

export default NavBar;
