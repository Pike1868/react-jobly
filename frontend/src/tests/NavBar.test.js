import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import NavBar from "../components/NavBar";
import { mockUserDetails } from "./commonSetup";

jest.mock("../context/UserContext", () => ({
  useUserContext: () => mockUserDetails,
}));

describe("Navbar", () => {
  it("renders without crashing", () => {
    render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    );
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
