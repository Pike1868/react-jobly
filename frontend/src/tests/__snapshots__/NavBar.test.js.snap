// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-1dnv1pm-MuiPaper-root-MuiAppBar-root"
  >
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
      href="/"
      style="text-decoration: none;"
    >
      <div
        class="MuiTypography-root MuiTypography-h4 css-go6ki8-MuiTypography-root"
        href="/"
      >
        JOBLY
      </div>
    </a>
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1b0ct7u-MuiToolbar-root"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-xdxgn1-MuiTypography-root-MuiLink-root"
        href="/companies"
        style="margin: 0px 10px;"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-1y942vo-MuiButtonBase-root-MuiButton-root"
          style="font-weight: 600;"
          tabindex="0"
          type="button"
        >
          Companies
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </a>
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-xdxgn1-MuiTypography-root-MuiLink-root"
        href="/jobs"
        style="margin: 0px 10px;"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-1y942vo-MuiButtonBase-root-MuiButton-root"
          style="font-weight: 600;"
          tabindex="0"
          type="button"
        >
          Jobs
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </a>
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-xdxgn1-MuiTypography-root-MuiLink-root"
        href="/profile"
        style="margin: 0px 10px;"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-1y942vo-MuiButtonBase-root-MuiButton-root"
          style="font-weight: 600;"
          tabindex="0"
          type="button"
        >
          Profile
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </a>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-1y942vo-MuiButtonBase-root-MuiButton-root"
        style="font-weight: 600; margin: 0px 10px;"
        tabindex="0"
        type="button"
      >
        Log Out
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
</DocumentFragment>
`;
